test/bdd/steps/network.js

```diff
@@ -466,6 +466,33 @@ Then(/^wallet and profile balances should diff by (\d+)$/, function (tokenDiff)
     expect(!!dc.state.newWalletBalance, 'newWalletBalance node not defined. Use other step to define it.').to.be.equal(true);
     expect(!!dc.state.oldWalletBalance, 'oldWalletBalance node not defined. Use other step to define it.').to.be.equal(true);
 
-    expect(dc.state.oldProfileBalance - dc.state.newProfileBalance, 'Profile diff should be equal to withdrawal amount').to.be.equal(tokenDiff);
-    expect(dc.state.newWalletBalance - dc.state.oldWalletBalance, 'Wallet diff should be equal to withdrawal amount').to.be.equal(tokenDiff);
+    expect(Math.abs(dc.state.oldProfileBalance - dc.state.newProfileBalance), 'Profile diff should be equal to withdrawal amount').to.be.equal(tokenDiff);
+    expect(Math.abs(dc.state.newWalletBalance - dc.state.oldWalletBalance), 'Wallet diff should be equal to withdrawal amount').to.be.equal(tokenDiff);
+});
+
+Given(/^I attempt to deposit (\d+) tokens from DC wallet*$/, { timeout: 120000 }, async function (tokenCount) {
+    // TODO expect tokenCount < walletBalance
+    expect(!!this.state.dc, 'DC node not defined. Use other step to define it.').to.be.equal(true);
+
+    const { dc } = this.state;
+    const host = dc.state.node_rpc_url;
+
+    await httpApiHelper.apiDeposit(host, tokenCount);
+});
+
+Then(/^Token deposit should be sucessfully completed from DC wallet$/, { timeout: 3000 }, async function () {
```

Comment:
High chance of race condition here. For example 'deposit-command-completed' can be fired long before calling dc.once(). This implementation should be a part of 'I attempt to deposit (\d+) tokens from DC wallet' sentence where you'd subscribe to events before actual call of the deposit API.